app/forms/questionnaire_form.py

```diff
@@ -66,10 +85,40 @@ def _get_target_total_and_currency(self, calculation, question):
         target_answer = self.schema.get_answer(calculation['answer_id'])
         return self.answer_store.filter(answer_ids=[target_answer['id']]).values()[0], target_answer.get('currency')
 
-    def _validate_date_range_question(self, question_id, period_from_id, period_to_id):
+    def validate_date_range_with_period_limits_and_single_date_limits(self, question_id, period_limits, date_from, date_to):  # pylint: disable=too-many-locals
+        if self._has_min_and_max_single_dates(date_from, date_to):
```

Comment:
This would be clearer if it was after [`if question['type'] == 'DateRange':`](https://github.com/ONSdigital/eq-survey-runner/pull/1521/files?diff=unified#diff-c5e43bea5b19fce9acef957e0a446c22R44) as both the methods that are then called below start with that check. You can then also remove it from  [`validate_date_range_with_single_date_limits`](https://github.com/ONSdigital/eq-survey-runner/pull/1521/files?diff=unified#diff-c5e43bea5b19fce9acef957e0a446c22R106). 
It's better practice to only call a method when needed and not to call it regardless and let the method decide if it should have been called.